### PR TITLE
[Infra] Switch zip building over to Xcode 15.2

### PR DESCRIPTION
**Pull request by ncooke3**
Per https://developer.apple.com/news/?id=fxu2qp7b, start building the zip with Xcode 15.2. Creating XCFrameworks on Xcode 15 versions before 15.3 has a bug where the `/var/` paths are not properly evaluated in their canonical form (`/private/var/`). To work around this, the `-create-xcframework` argument is given canonical paths. See https://github.com/firebase/firebase-ios-sdk/pull/12737#issuecomment-2053645735.

### CODE REVIEW

File: .github/workflows/zip.yml
Comment (ncooke3) [inline .github/workflows/zip.yml:1]: Will need a changelog entry before merging.

**Comment by ncooke3** [comment]
Building with 15.2 has a bug that was fixed in 15.3: 
```
Fixed a bug where creating an xcframework from a framework or library located at a path containing a symlink (such as /var which is a symlink to /private/var) would fail. (115786062) (FB13191683)
```

Which I believe is causing failure (https://github.com/firebase/firebase-ios-sdk/actions/runs/8620842172/job/23628537213?pr=12737#step:7:19446). 

I _think_ one idea to workaround would be to add an option to the zip builder to _not_ use a temporary directory (as it contains the problematic symlink) and instead use a directory passed in (e.g. ~/path/to/build/intermediates`). I will investigate more tomorrow.

**Comment by ncooke3** [comment]
Locally reproducible with
```
➜  firebase git:(main) sudo xcode-select --switch /Applications/Xcode_15.2.app 
➜  firebase git:(main) swift run --package-path ReleaseTooling zip-builder \
  --disable-carthage-version-check \
  --no-dynamic \
  --keep-build-artifacts \
  --no-update-pod-repo \
  --pods FirebaseSharedSwift --platforms ios --no-include-catalyst
<-- snip -->
Successfully built FirebaseSharedSwift for iphonesimulator. Build log is at file:///var/folders/6h/fgkrlkjs0ygdzpbhyr9l4k1c0000gn/T/ZipRelease/2024-04-12T17-06-31/build_logs/FirebaseSharedSwift-iphonesimulator.txt.
buildDir: file:///var/folders/6h/fgkrlkjs0ygdzpbhyr9l4k1c0000gn/T/ZipRelease/2024-04-12T17-06-31/project-ios/iphonesimulator
buildSliced returns file:///var/folders/6h/fgkrlkjs0ygdzpbhyr9l4k1c0000gn/T/ZipRelease/2024-04-12T17-06-31/project-ios/iphonesimulator/Release-iphonesimulator/FirebaseSharedSwift/FirebaseSharedSwift.framework/
Building file:///var/folders/6h/fgkrlkjs0ygdzpbhyr9l4k1c0000gn/T/ZipRelease/2024-04-12T17-06-31/xcframeworks/FirebaseSharedSwift.xcframework with command:
/usr/bin/xcodebuild -create-xcframework -framework /var/folders/6h/fgkrlkjs0ygdzpbhyr9l4k1c0000gn/T/ZipRelease/2024-04-12T17-06-31/platform_frameworks/iphonesimulator/FirebaseSharedSwift.framework -framework /var/folders/6h/fgkrlkjs0ygdzpbhyr9l4k1c0000gn/T/ZipRelease/2024-04-12T17-06-31/platform_frameworks/iphoneos/FirebaseSharedSwift.framework -output /var/folders/6h/fgkrlkjs0ygdzpbhyr9l4k1c0000gn/T/ZipRelease/2024-04-12T17-06-31/xcframeworks/FirebaseSharedSwift.xcframework
error: cannot compute path of binary 'Path(str: "/private/var/folders/6h/fgkrlkjs0ygdzpbhyr9l4k1c0000gn/T/ZipRelease/2024-04-12T17-06-31/platform_frameworks/iphonesimulator/FirebaseSharedSwift.framework/FirebaseSharedSwift")' relative to that of '/var/folders/6h/fgkrlkjs0ygdzpbhyr9l4k1c0000gn/T/ZipRelease/2024-04-12T17-06-31/platform_frameworks/iphonesimulator/FirebaseSharedSwift.framework'
ZipBuilder/FrameworkBuilder.swift:727: Fatal error: Could not build xcframework for FirebaseSharedSwift exit code 70: 
[1]    82988 trace trap  swift run --package-path ReleaseTooling zip-builder  --no-dynamic   --pods  
```

**Comment by ncooke3** [comment]
path [URL]: `file:///var/folders/6h/fgkrlkjs0ygdzpbhyr9l4k1c0000gn/T/`
path (resolving symlinks) [URL]: `file:///var/folders/6h/fgkrlkjs0ygdzpbhyr9l4k1c0000gn/T/`
path (canonical path) [String]: `/private/var/folders/6h/fgkrlkjs0ygdzpbhyr9l4k1c0000gn/T`

File: .github/workflows/zip.yml
Comment (ncooke3) [reply .github/workflows/zip.yml:1]: [6739980](https://github.com/firebase/firebase-ios-sdk/pull/12737/commits/6739980ccd8ae605897bf582c78a272ba2ad3623)

File: ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift
Comment (paulb777) [inline ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift:717]: Good find! Should we catch and log a good error instead of force unwrapping and `try!`?

**Comment by ncooke3** [comment]
Perhaps we have to hold off on building the zip with the ARM mac. It took over 24 hrs for the original run to pick a machine, and I don't think it ever did before timing out. I kicked if off again this morning, but it still hasn't been picked up in the 90 minutes since I re-kicked it off.

**Comment by paulb777** [comment]
Choosing `macos-14` seems a reliable way to get arm macs

**Comment by ncooke3** [comment]
There must have been something wrong with [83ff298](https://github.com/firebase/firebase-ios-sdk/pull/12737/commits/83ff298490a180a16536328a2ad863bf8b372759) because CI did not run as expected. Trying again with the `-arm` suffix removed.

**Comment by ncooke3** [comment]
Testing the macos-14 built artifact on the quickstart tests: https://github.com/firebase/firebase-ios-sdk/actions/runs/8707424883/job/23882445114

**Comment by ncooke3** [comment]
By bumping the macOS version of the quickstart runner, we're running into https://github.com/firebase/firebase-ios-sdk/blob/main/.github/workflows/zip.yml#L463

Looking into ways we can fix the FirebaseUI XCFrameworks...

**Comment by paulb777** [comment]
Finding an Xcode 15 build solution for Firebase UI sounds good - but I'm fine with a workaround for this PR.

**Comment by ncooke3** [comment]
I think I'm going to need to comment out the Firestore test. My workaround around for now was to move the resource bundles that were embedded in the `FirebaseAuth*UI.xcframework` into the platform level `.frameworks`. I cannot reproduce the exact CI failure locally on either an Intel or arm mac.

**Review by andrewheard — APPROVED** [review]
Thanks!

**Review by paulb777 — APPROVED** [review]

**Pull request merged**
